Core/GDCore/Project/Variable.h

```diff
@@ -26,18 +28,33 @@ namespace gd {
  */
 class GD_CORE_API Variable {
  public:
+  enum TYPES {

```

Comment:
Nitpicking: enum names should usually be singular. Here it's `TYPE`. This is because each value represents "a" type.
See if you can use `enum class` - I've not done it in the rest of the codebase but it might gives better scoping.

Finally, I think we don't need UPPER_CASING here, at least I've seen usage of PascalCase for both the name (`Type`) and the content (`String`, `Number`, etc...) in the codebase, and because an enum is a type, it seems a bit too much to use UPPER_CASE (which is more for constants).

Reply:
`enum class` has a problem. The WebIDL binder accepts them, but doesn't store them in `gd.Variable` but in `gd` directly. This is problematic and pollutes `gd`. I'm staying on "classic" `enum` for now.

Reply:
Sure no worries then!